src/main/kotlin/javabot/dao/impl/openweathermap/OpenWeatherMapHandler.kt

```diff
@@ -0,0 +1,77 @@
+package javabot.dao.impl.openweathermap
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import javabot.JavabotConfig
+import javabot.dao.WeatherHandler
+import javabot.dao.impl.openweathermap.model.OWWeather
+import javabot.model.Weather
+import net.iakovlev.timeshape.TimeZoneEngine
+import org.apache.http.client.fluent.Request
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.xml.sax.InputSource
+import java.net.URL
+import java.net.URLEncoder
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+/**
+ * Implements a weather service Dao using OpenWeather's weather API.
+ * We may need to eventually think about OWM's terms of service; the free key allows
+ * only 1000 requests a day. This may not be a problem in normal usage, but what
+ * if a malicious bot or user issues 4000 weather requests? They'd get denied (the API
+ * would reject them) but it's still impolite behavior on the part of the bot.
+ *
+ * @see javabot.dao.impl.WeatherDao
+ */
+class OpenWeatherMapHandler : WeatherHandler {
+    override fun getWeatherFor(place: String, javabotConfig: JavabotConfig): Weather? {
+        val mapper = ObjectMapper()
+        val apiKey = javabotConfig.openweathermapToken()
+        return if (apiKey.isNotEmpty()) {
+            try {
+                val location = place.replace(" ", "+")
+                val url = "$API_URL$location&APPID=$apiKey"
+
+                val weatherResponse = Request
+                        .Get(url)
+                        .execute()
+                        .returnContent().asString()
+                val data = mapper.readValue<OWWeather>(weatherResponse, OWWeather::class.java)
+
+                val weather = Weather()
+                weather.city = "${data.name} (${data.sys!!.country})"
+                weather.humidity = data.main!!.humidity.toString()
```

Comment:
this feels dirty but I'm not sure I care that much...

Reply:
There's another issue related to how the Weather object is constructed. Yes, that's ... gross.